src/main/kotlin/com/dropbox/mobile/hypershard/Hypershard.kt

```diff
@@ -202,20 +218,59 @@ class RealHyperShard(
         }
         return tests
     }
-}
 
-fun main(args: Array<String>) {
-    checkUsage(args)
-    val annotationName = args[0]
-    val dirs = args.sliceArray(1 until args.size)
-    val processor = RealHyperShard(annotationName, dirs)
-    processor.gatherTests().stream().forEach(::println)
+    /**
+     * Returns true if we found the class annotation or if's meant to be empty
+     */
+    private fun shouldProcessTestMethods(
+        ktClass: KtClass,
+        annotationValue: ClassAnnotationValue
+    ): Boolean {
+        return when (annotationValue) {
+            is ClassAnnotationValue.Present -> {
+                var found = false
+                for (annotationEntry in ktClass.annotationEntries) {
+                    if (annotationEntry.shortName.toString() == annotationValue.annotationName) {
+                        found = true
+                        break
+                    }
+                }
+                found
+            }
+            is ClassAnnotationValue.Empty -> true
+        }
+    }
 }
 
-private fun checkUsage(dirs: Array<String>) {
-    check(dirs.isNotEmpty()) {
-        "Usage: " +
-            "java -jar hypershard-x.y.z-all.jar " +
-            "annotationName dir1 dir2 dir3 ..."
+/**
+ * Entry point for Hypershard CLI
+ */
+class HypershardCommand :
+    CliktCommand(
```

Comment:
name feels weird, looks like "clicked", maybe capital K or rename a bit?

Reply:
`ClicktCommand` is from a 3rd party library